apps/admin/backend/src/cvr_files.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  FileSystemEntryType,
+  listDirectoryOnUsbDrive,
+} from '@votingworks/backend';
+import { throwIllegalValue } from '@votingworks/basics';
+import { LogEventId, Logger } from '@votingworks/logging';
+import { ElectionDefinition } from '@votingworks/types';
+import {
+  generateElectionBasedSubfolderName,
+  parseCvrFileInfoFromFilename,
+  SCANNER_RESULTS_FOLDER,
+} from '@votingworks/utils';
+import { join } from 'path';
+
+/**
+ * Metadata about a found cast vote record file.
+ */
+export interface CastVoteRecordFileMetadata {
+  readonly name: string;
+  readonly path: string;
+  readonly cvrCount: number;
+  readonly scannerIds: readonly string[];
+  readonly exportTimestamp: Date;
+  readonly isTestModeResults: boolean;
+}
+
+/**
+ * Gets the metadata, including the path, of cast vote record files found in
+ * the default location on a mounted USB drive. If there is no mounted USB
+ * drive or the USB drive appears corrupted then it will return an empty array.
+ */
+export async function listCastVoteRecordFilesOnUsb(
+  electionDefinition: ElectionDefinition,
+  logger: Logger
+): Promise<CastVoteRecordFileMetadata[]> {
+  const { election, electionHash } = electionDefinition;
+  const fileSearchResult = await listDirectoryOnUsbDrive(
+    join(
+      SCANNER_RESULTS_FOLDER,
+      generateElectionBasedSubfolderName(election, electionHash)
+    )
+  );
+
+  if (fileSearchResult.isErr()) {
+    const errorType = fileSearchResult.err().type;
+    switch (errorType) {
+      case 'no-entity':
+        await logger.log(LogEventId.CvrFilesReadFromUsb, 'system', {
+          message:
+            'No cast vote record files automatically found on USB drive. User is allowed to manually select files.',
+          disposition: 'success',
+        });
+        break;
+      case 'not-directory':
+      case 'permission-denied':
+        await logger.log(LogEventId.CvrFilesReadFromUsb, 'system', {
+          message:
+            'Error accessing cast vote record files on USB drive, which may be corrupted.',
+          disposition: 'failure',
+        });
+        break;
+      case 'no-usb-drive':
+      case 'usb-drive-not-mounted':
+        // we're just polling without a USB drive in these cases, no issue
+        break;
+      default:
+        throwIllegalValue(errorType);
+    }
+
+    return [];
```

Comment:
If we're returning `[]` in all of these cases, how does the app know to differentiate between the corrupted USB case (where it should prob show an error to the user) and the other cases where they can manually select a file?

Reply:
Formerly, if one of these errors happen an error would just be thrown to the error boundary. 

Now, I'm just ignoring it. Despite the logging, the user is still allowed to manually select a file in this case. Although they will likely hit another problem with that USB drive, it's possible they do not. 

I mostly didn't care about this case because I did not find it worth building a user error message for. In order to get either the permission denied or the not directory error, the user would have had to manipulate the file system manually at another computer rather than export from one of our machines. We don't want them doing that anyway.